src/schema/v1.json

```diff
@@ -3,63 +3,121 @@
-   "name",
-   "ostree-commit",
-   "ostree-content-bytes-written",
-   "ostree-content-checksum",
-   "ostree-n-cache-hits",
-   "ostree-n-content-total",
-   "ostree-n-content-written",
-   "ostree-n-metadata-total",
-   "ostree-n-metadata-written",
-   "ostree-timestamp",
-   "ostree-version",
-   "rpm-ostree-inputhash",
-   "summary"
-  ],
- "optional": [
-   "aliyun",
-   "amis",
-   "azure",
-   "build-url",
-   "gcp",
-   "oscontainer",
-   "pkgdiff",
-   "coreos-assembler.container-config-git",
-   "coreos-assembler.container-image-git"
-  ],
+ "title":"CoreOS Assember v1 meta.json schema",
+ "allOf": [
+     {
+         "required": [
+             "buildid",
+             "name",
+             "ostree-commit",
+             "ostree-content-bytes-written",
+             "ostree-content-checksum",
+             "ostree-n-cache-hits",
+             "ostree-n-content-total",
+             "ostree-n-content-written",
+             "ostree-n-metadata-total",
+             "ostree-n-metadata-written",
+             "ostree-timestamp",
+             "ostree-version",
+             "rpm-ostree-inputhash",
+             "summary"
+         ],
+         "optional": [
+           "images",
+           "aliyun",
+           "amis",
+           "azure",
+           "build-url",
+           "gcp",
+           "oscontainer",
+           "pkgdiff",
+
+           "coreos-assembler.basearch",
+           "coreos-assembler.build-timestamp",
+           "coreos-assembler.code-source",
+           "coreos-assembler.config-dirty",
+           "coreos-assembler.config-gitrev",
+           "coreos-assembler.container-config-git",
+           "coreos-assembler.container-image-git",
+           "coreos-assembler.image-config-checksum",
+           "coreos-assembler.image-genver",
+           "coreos-assembler.image-input-checksum",
+           "fedora-coreos.parent-commit",
+           "fedora-coreos.parent-version",
+           "ref"
+          ]
+     },
+     {
+         "if": {
+             "properties": {
+                 "name": { "const": "fedora-coreos"  }
+             }
+         },
+         "then": {
+                "required": [
+                   "ref"
+                  ],
+                "optional": [
+                  "fedora-coreos.parent-commit",
+                  "fedora-coreos.parent-version"
+                ]
+         },
+         "else": {}
+    },
+    {
+         "if": {
+             "properties": {
+                 "name": { "pattern": "^(rhcos|fedora-coreos)"  }
+             }
+         },
+         "then": {
+                "required": [
+                   "coreos-assembler.basearch",
+                   "coreos-assembler.build-timestamp",
+                   "coreos-assembler.code-source",
+                   "coreos-assembler.config-dirty",
+                   "coreos-assembler.config-gitrev",
+                   "coreos-assembler.container-config-git",
+                   "coreos-assembler.container-image-git",
```

Comment:
Let's not make these required again please?  I don't see a connection between "custom coreos-like builds" and "custom coreos-assembler builds inside an existing development container".

IOW, I want to be able to test patches to coreos-assembler and rpm-ostree inside my dev container, targeting all of fcos/rhcos *and* e.g. [FCOSB](https://github.com/cgwalters/fedora-silverblue-config).

Reply:
The problem is that they _are_ required for RHCOS and the purpose of the schema is to validate the results from builds. I am sympathetic towards your pain @cgwalters -- I am -- but your argument is a required production schema should be weakened for you as a developer based on your workflow. 

https://github.com/coreos/coreos-assembler/pull/1089 was targeted at helping you out with this. 

But the larger problem, IMO, is that we have developers who run their own pet container, while COSA expects certain things; that's the tension here. I'm open to ideas

Reply:
I think delivering coreos-assembler as a container image has been very successful; I often tell people about that.

BUT I don't think we should entirely rule out e.g. shipping it as an RPM too.

And conceptually, these keys only cover the container build path.

And further, they duplicate metadata already available in the container metadata (OpenShift builds e.g. inject the git commit); unfortunately quay.io's build system is more primitive.

So I'm not sure I'd agree these keys are "required for RHCOS".

I guess I can just set the environment variable for now, it'll just be annoying.

Reply:
I'll also chime in as a developer who hacks on cosa from his pet container. :dog2: I think it's important to cover that use-case. Not just for rpm-ostree/`$util`, but just overall I think it yields a faster edit-build-test feedback loop which makes hacking more pleasant. I've documented it [in the README](https://github.com/coreos/coreos-assembler#installing-cosa-inside-an-existing-container) too.

Reply:
Dropped in the last push. 

After(if) this merged I'll through up one that allows for per-recipe overrides.